### ISSUE TITLE
error after upgrade

### ISSUE THREAD

**Issue by MHagreis** [moroen/IKEA-Tradfri-plugin#120]
2020-11-03 15:42:17.194 Error: (IKEA-Tradfri) ----> Line 773 in '/home/pi/domoticz/plugins/IKEA-Tradfri/plugin.py', function onHeartbeat
2020-11-03 15:42:17.194 Error: (IKEA-Tradfri) ----> Line 730 in '/home/pi/domoticz/plugins/IKEA-Tradfri/plugin.py', function onHeartbeat
2020-11-03 15:42:27.164 Error: (IKEA-Tradfri) 'onHeartbeat' failed 'NameError':'name 'close_connection' is not defined'.
2020-11-03 15:42:27.164 Error: (IKEA-Tradfri) ----> Line 773 in '/home/pi/domoticz/plugins/IKEA-Tradfri/plugin.py', function onHeartbeat
2020-11-03 15:42:27.164 Error: (IKEA-Tradfri) ----> Line 730 in '/home/pi/domoticz/plugins/IKEA-Tradfri/plugin.py', function onHeartbeat
2020-11-03 15:42:36.016 (Buienradar) General/Custom Sensor (Next Rainshower Leadtime)
2020-11-03 15:42:37.186 Error: (IKEA-Tradfri) 'onHeartbeat' failed 'NameError':'name 'close_connection' is not defined'.
2020-11-03 15:42:37.186 Error: (IKEA-Tradfri) ----> Line 773 in '/home/pi/domoticz/plugins/IKEA-Tradfri/plugin.py', function onHeartbeat
2020-11-03 15:42:37.186 Error: (IKEA-Tradfri) ----> Line 730 in '/home/pi/domoticz/plugins/IKEA-Tradfri/plugin.py', function onHeartbeat

**Comment by MHagreis** [comment]
fixed by re-installing setting key again and renaming the tradfricoap in the IKEA-Tradfri folder to  tradfricoap-old (later will remove it :-) )

**Comment by Bolten88** [comment]
Hello @MHagreis i receive these errors as well. Can you describe more detailed what you did to resolve it?

**Comment by moroen** [comment]
@Bolten88 

Have you removed the tradfricoap directory from the plugin?

**Comment by Bolten88** [comment]
> @Bolten88
> 
> Have you removed the tradfricoap directory from the plugin?

Yes, i did that. That was one of the action with the upgrade right?

![Knipsel](https://user-images.githubusercontent.com/71372668/98354373-e81a1d80-2020-11eb-85fb-809597301241.PNG)

**Comment by MHagreis** [comment]
I did that that helped but also inserted the key once more via config (hub key)
with the following command 
$ python3 plugin.py config gw IP KEY

**Comment by Bolten88** [comment]
> I did that that helped but also inserted the key once more via config (hub key)
> with the following command
> $ python3 plugin.py config gw IP KEY

I did that a few days ago allready because i added a new IKEA wall plug to the system. But i will do that again to be sure and see if the error returns or not. I see the error like once a day so i will look for it tomorrow.

**Comment by moroen** [comment]
@Bolten88 

Looking closer at the error, you definitely have a different error than MHagreis! The plugin appears to chocke on a unknown reply from the gateway, I suspect it might be timeout of some sort. I'll take closer look at it, but I have understood correctly that the plugin otherwise works?

**Comment by Bolten88** [comment]
@moroen i figured out what is causing the error. It is sort of a human error thing ;-).

There was a small change in my network. I connected the IKEA gateway directly to my router. This router restarts every night while i'm asleep. Offcourse during this reboot the gateway is not reachable and domoticz/plugin is complaining about that. I matched the times of the error witch the time of the reboot. I plugged the gateway back to the switch where domoticz is connected to and then the error did not return anymore.
